components/gitpod-protocol/src/protocol.ts

```diff
@@ -293,11 +293,25 @@ export namespace EnvVar {
             return res;
         }
 
-        (imageAuth.value || "")
-            .split(",")
-            .map((e) => e.trim().split(":"))
-            .filter((e) => e.length == 2)
-            .forEach((e) => res.set(e[0], e[1]));
+        (imageAuth.value || "").split(",").forEach((entry) => {
```

Comment:
Somehow it's very hard for me to understand the new implementation, while the old I could grok in an instant.

Does it really need that many lines? And if the answer is really yes: We should probably have a separate (in-function?) "parse" method.

Reply:
I introduced an inline function and reduced lines of code in c0be79f32f0bd5066843e5df19a4bbb81e2f8aa8.

You'll notice a truthiness check on all parts, this is to ensure we're not working with junk. I could remove it, but that'll cause tests to fail like so:

<img width="1701" alt="image" src="https://github.com/user-attachments/assets/ac761268-6f4c-4e7c-ace3-66de674de3a7" />

LMKWYT? I'll move onto checking other components now.